pulp_smash/tests/rpm/api_v2/test_upload_publish.py

```diff
@@ -13,19 +13,138 @@
 
 from itertools import product
 
-from pulp_smash import api, utils
+from pulp_smash import api, selectors, utils
 from pulp_smash.compat import urljoin
 from pulp_smash.constants import (
     CALL_REPORT_KEYS,
     CONTENT_UPLOAD_PATH,
     REPOSITORY_PATH,
     RPM,
+    RPM_FEED_URL,
     RPM_URL,
 )
-from pulp_smash.tests.rpm.api_v2.utils import gen_distributor, gen_repo
+from pulp_smash.tests.rpm.utils import gen_erratum
+from pulp_smash.tests.rpm.api_v2.utils import (
+    gen_distributor,
+    gen_repo,
+    get_repomd_xml,
+)
 from pulp_smash.tests.rpm.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
 
 
+def parse_updateinfo_update(update):
+    """Create an erratum from an ``updateinfo.xml`` update.
+
+    This function creates an empty erratum, then parses a top-level ``update``
+    section from an ``updateinfo.xml`` file, and populates the empty erratum
+    from such.
+
+    .. WARNING:: There is no schema for ``updateinfo.xml`` files. Instead, this
+        function's parsing logic is reverse-engineered from the DNF source
+        code. As a result, this function may exhibit erroneous behaviour.
+
+        As of this writing, the DNF source code can be cloned from
+        ``https://github.com/rpm-software-management/dnf.git``. The files used
+        by this function are:
+
+        * `dnf/cli/commands/updateinfo.py`_
+        * `tests/repos/rpm/updateinfo.xml`_
+
+    .. NOTE:: DNF is incompatible with the openSUSE ``updateinfo.xml`` schema.
```

Comment:
Thanks!

Research was a huge part of this PR. It took a while to find out where errata come from and what they should look like, and where `updateinfo.xml` files go and what they should look like.